### PR TITLE
Make core.get_favorites("local") ignore duplicates

### LINKED ISSUE #4729: Crash on tab_multiplayer.lua and tab_online.lua: Favourites duplicate
Happened loading the multiplayer menu after logging out from a server:

```
2016-11-04 19:31:57: ERROR[Main]: GUIEngine: execution of menu script failed: Failed to load and run script from /home/sofar/git/minetest/bin/../builtin/init.lua:
2016-11-04 19:31:57: ERROR[Main]: ...git/minetest/bin/../builtin/mainmenu/tab_multiplayer.lua:68: attempt to index a nil value
2016-11-04 19:31:57: ERROR[Main]: stack traceback:
2016-11-04 19:31:57: ERROR[Main]: 	...git/minetest/bin/../builtin/mainmenu/tab_multiplayer.lua:68: in function 'get_formspec'
2016-11-04 19:31:57: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/fstk/tabview.lua:74: in function 'get_formspec'
2016-11-04 19:31:57: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/fstk/ui.lua:107: in function 'update'
2016-11-04 19:31:57: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/mainmenu/init.lua:164: in function 'init_globals'
2016-11-04 19:31:57: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/mainmenu/init.lua:169: in main chunk
2016-11-04 19:31:57: ERROR[Main]: 	[C]: in function 'dofile'
2016-11-04 19:31:57: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/init.lua:44: in main chunk
2016-11-04 19:31:57: ERROR[Main]: No future without mainmenu
```
Aborted (core dumped)

### PR DESCRIPTION
**Pull request by pauloue**
Fixes #4729

### CODE REVIEW

**Comment by SmallJoker** [comment]
Thanks for the PR - but is this the right way to solve the issue? If there are duplicated items, does `insert` not work properly?

**Comment by pauloue** [comment]
`insert` works correctly, I don't think it's possible that duplicates get added to the file, so maybe this is a non-issue. But it's probably still worth avoiding a crash in the case that the file is manually edited to contain duplicates. Also `delete_favorite` already checks for duplicates, so why do the same in `get_favorite`.

File: src/serverlist.cpp [hunk 1]
```diff
@@ -155,6 +155,16 @@ std::vector<ServerListSpec> deSerialize(const std::string &liststring)
 			server["address"] = tmp;
 			std::getline(stream, tmp);
 			server["port"] = tmp;
+			bool unique = true;
+			for (ServerListSpec &added : serverlist) {
```
Comment (nerzhul) [inline src/serverlist.cpp]: this should be const we don't  modify the list objects

Comment (pauloue) [reply src/serverlist.cpp]: done

**Comment by paramat** [comment]
Blocker for 5.0.0?

**Comment by nerzhul** [comment]
It can be delayed, it's just a historical dup fix

**Pull request merged**